comp/rdnsquerier/impl/rdnsquerier_test.go

```diff
@@ -986,3 +993,81 @@ func TestCachePersist(t *testing.T) {
 	})
 	ts.validateExpected(t, expectedTelemetry)
 }
+
+func TestGetHostnameSync(t *testing.T) {
+	overrides := map[string]interface{}{
+		"network_devices.netflow.reverse_dns_enrichment_enabled": true,
+	}
+
+	ts := testSetup(t, overrides, true, nil, 0)
+	internalRDNSQuerier := ts.rdnsQuerier.(*rdnsQuerierImpl)
+
+	// Vic to reformat this test
+	// tests := []struct {
+	// 	description string
+	// 	ip          string
+	// 	expected    string
+	// 	errMsg      string
+	// }{
+	// 	{},
+	// }
+
+	ctx, cancel := context.WithTimeout(ts.ctx, 100*time.Second)
```

Comment:
It's probably fine to use the same context across these, but if for some reason there were to be some lock contention or something, it could take an extra 100s to fail the test suite. It's a bit tedious but maybe you could do it for each one?

This is where something like table tests can be super useful because you only have to write the test logic once, the rest of the work is just calibrating the inputs. https://dave.cheney.net/2019/05/07/prefer-table-driven-tests